src/pybind/mgr/rook/module.py

```diff
@@ -639,3 +684,43 @@ def upgrade_status(self) -> orchestrator.UpgradeStatusSpec:
     @handle_orch_error
     def upgrade_ls(self, image: Optional[str], tags: bool, show_all_versions: Optional[bool]) -> Dict[Any, Any]:
         return {}
+
+    # Retry decorator for handling transient Kubernetes API failures
+    @retry(on_exception=ApiException, tries=7, delay=1, backoff=2, max_delay=60)
+    def fetch_k8s_secret(self, secret_name: str) -> Optional[V1Secret]:
```

Comment:
after most recent push, it's now complaining about this line
```
File "/home/docs/checkouts/readthedocs.org/user_builds/ceph/checkouts/61468/src/pybind/mgr/rook/module.py", line 690, in RookOrchestrator
    def fetch_k8s_secret(self, secret_name: str) -> Optional[V1Secret]:
NameError: name 'V1Secret' is not defined
```

Reply:
I have updated the PR again with fixes. All tests passed except [ceph rook orchestrator e2e tests](https://jenkins.ceph.com/job/ceph-orchestrator-rook-e2e/1398/). I could find relevancy to my changes from logs.